tsdb/head_wal.go

```diff
@@ -438,10 +438,9 @@ func (wp *walSubsetProcessor) waitUntilIdle() {
 	}
 	wp.input <- []record.RefSample{}
```

Comment:
Unrelated nitpick: this would be _better_:
```suggestion
	wp.input <- nil
```

Reply:
Also, it's unclear here why this is pushed. I understand that it's pushed just to block the for loop in the running position until the actual input comes in.

Reply:
We push an input then watch for the input queue to reach zero length. 
No-one else is pushing to the input, so once the one we pushed has been consumed we know the worker is doing no real work, which is equivalent to “idle”.